### PR TITLE
Node10 support

### PR DESCRIPTION
**Pull request by devsatishm**
### Description of the Change

- Removed support for node6 & python2
- Added support for node10
- Renamed template folder names (`lambda` -> `function`)

### CODE REVIEW

File: builds/jazz-build-module/sls-app/serverless-build-rules.yml [hunk 1]
```diff
@@ -70,13 +70,13 @@ provider:
     sbr-constraint:
       sbr-enum:
         - nodejs8.10
+        - nodejs10.x
```
Comment (raghits) [inline builds/jazz-build-module/sls-app/serverless-build-rules.yml]: is `nodejs10.x` a valid option - I mean the `.x`?

Comment (devsatishm) [reply builds/jazz-build-module/sls-app/serverless-build-rules.yml]: Yep - https://docs.aws.amazon.com/en_pv/lambda/latest/dg/programming-model.html

File: builds/service-onboarding-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -90,7 +90,7 @@ node  {
             } else {
```
Comment (raghits) [inline builds/service-onboarding-build-pack/Jenkinsfile]: This block needs to be refined. For later!

File: builds/jenkins-build-pack-api/Jenkinsfile [hunk 3]
```diff
@@ -899,7 +899,7 @@ def validateDeploymentConfigurations(def prop) {
       error "Wrong configuration. Value for Key 'providerRuntime' is missing in the configuration"
 
     } else {
-      def validRuntimes = ["nodejs4.3", "nodejs6.10", "nodejs8.10", "python2.7", "python3.6", "java8", "go1.x"]
+      def validRuntimes = ["nodejs8.10", "nodejs10.x", "python3.6", "java8", "go1.x"]
```
Comment (raghits) [inline builds/jenkins-build-pack-api/Jenkinsfile]: Will this break build for services older runtimes?

File: builds/jenkins-build-pack-function/Jenkinsfile [hunk 4]
```diff
@@ -716,7 +715,7 @@ def validateDeploymentConfigurations(def prop) {
 		if (_runtime == "") {
 			error "Wrong configuration. Value for Key 'providerRuntime' is missing in the configuration"
 		} else {
-			def validRuntimes = ["nodejs4.3", "nodejs6.10", "nodejs8.10", "python2.7", "python3.6", "java8", "go1.x", "c#"]
+			def validRuntimes = ["nodejs8.10", "nodejs10.x", "python3.6", "java8", "go1.x", "c#"]
```
Comment (raghits) [inline builds/jenkins-build-pack-function/Jenkinsfile]: Will this break build for services older runtimes?

File: builds/jenkins-build-pack-sls-app/Jenkinsfile [hunk 5]
```diff
@@ -632,7 +632,7 @@ def validateDeploymentConfigurations(def prop) {
 		if (_runtime == "") {
 			error "Wrong configuration. Value for Key 'providerRuntime' is missing in the configuration"
 		} else {
-			def validRuntimes = ["nodejs6.10", "nodejs8.10", "python2.7", "python3.6", "java8", "go1.x"]
+			def validRuntimes = ["nodejs8.10", "nodejs10.x", "python3.6", "java8", "go1.x"]
```
Comment (raghits) [inline builds/jenkins-build-pack-sls-app/Jenkinsfile]: Will this break build for services older runtimes?

Comment (devsatishm) [reply builds/jenkins-build-pack-sls-app/Jenkinsfile]: Not really. At this point, Jazz doesn't support stack upgrades completely (which means there won't be any "old"  services when the stack starts).

File: builds/jenkins-build-pack-function/Jenkinsfile (hunk 4 above)
Comment (devsatishm) [reply builds/jenkins-build-pack-function/Jenkinsfile]: Addressed below.

File: builds/jenkins-build-pack-api/Jenkinsfile (hunk 3 above)
Comment (devsatishm) [reply builds/jenkins-build-pack-api/Jenkinsfile]: Addressed below.

File: builds/jenkins-build-pack-sls-app/Jenkinsfile (hunk 5 above)
Comment (raghits) [reply builds/jenkins-build-pack-sls-app/Jenkinsfile]: But I guess the same is not true with Internal?

**Pull request merged**